src/type/preload.ts

```diff
@@ -640,3 +644,12 @@ export interface MessageBoxReturnValue {
 }
 
 export const SandboxKey = "electron" as const;
+
+/** プロジェクト一時ファイル */
+export type TempProjectType = {
+  appVersion?: string;
+  audioKeys?: AudioKey[];
+  audioItems?: Record<AudioKey, AudioItem>;
+  projectFilePath?: string;
+  commandStoreState?: CommandStoreState;
+};
```

Comment:
全部optionalにするよりUnion Typeを使った方がいいと思います。（project.tsの607行目のようなアサーションも減らせますし）
```suggestion
/** プロジェクト一時ファイル */
export type TempProjectType = {
  // このキーの名前は変えてもいいかも
  state: "editing";
  appVersion: string;
  audioKeys: AudioKey[];
  audioItems: Record<AudioKey, AudioItem>;
  projectFilePath: string;
  commandStoreState: CommandStoreState;
} | {
  state: "saved"
  projectFilePath: string;
} | {
  state: "none";
}
```

Reply:
全部optionalにしないのは賛成です！

ちょっと別の提案として、そもそもプロジェクトファイルを保存した場合でも異常終了時と同様にワークスペース全ての状態を復元する挙動を提案してみました。
https://github.com/VOICEVOX/voicevox/pull/1438#pullrequestreview-1742114025
この場合だとUnion Typeはいらないかもです！

Reply:
常にコマンドを復元する仕様でも一時ファイルの状態を見て異常終了/正常終了を見分けるようにする必要があったので、以下のような型に変更しました。

```typescript
export type TempProjectType =
  | {
      /** プロジェクトの一部・またはすべてが未保存の状態 */
      state: "unSaved";
      project: ProjectType;
      projectFilePath?: string;
      commandStoreState: CommandStoreState;
    }
  | {
      /** プロジェクトが保存されている状態 */
      state: "saved";
      projectFilePath: string;
      commandStoreState: CommandStoreState;
    }
  | {
      /** プロジェクト一時ファイルの初期状態 */
      state: "none";
    };
```

`state: "unSaved"` の `projectFilePath` がオプショナルなのは、新規作成後に一度も保存していない状態を想定しているためです。

Reply:
良い感じなのかなと思いました！！
いったん元のコメントは解決されたかなと思うのでResolveさせていただきます！